### PR TITLE
Add VirtualHubBgpConnections, VirtualHubIpConfigurations, NetworkVirtualApplianceSites child resources

### PR DESCRIPTION
**Pull request by naveenchekuri**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json [hunk 1]
```diff
@@ -3413,183 +3413,651 @@
...
-        "tags": [
-          "ExpressRouteConnections"
-        ],
-        "operationId": "ExpressRouteConnections_Get",
+      "put": {
+        "operationId": "VirtualHubIpConfiguration_CreateOrUpdate",
         "x-ms-examples": {
-          "ExpressRouteConnectionGet": {
-            "$ref": "./examples/ExpressRouteConnectionGet.json"
+          "VirtualHubRouteTableV2Put": {
```
Comment (MyronFanQiu) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json:3942]: Should it be VirtualHubIpConfigurationPut?

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json [hunk 2]
```diff
@@ -3413,183 +3413,651 @@
...
       "delete": {
-        "tags": [
-          "ExpressRouteConnections"
-        ],
-        "operationId": "ExpressRouteConnections_Delete",
+        "operationId": "VirtualHubIpConfiguration_Delete",
         "x-ms-examples": {
-          "ExpressRouteConnectionDelete": {
-            "$ref": "./examples/ExpressRouteConnectionDelete.json"
+          "VirtualHubRouteTableV2Delete": {
```
Comment (MyronFanQiu) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json:4013]: same here

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json [hunk 3]
```diff
@@ -4204,10 +4672,164 @@
...
+          "type": "string",
+          "description": "The private IP address of the IP configuration."
+        },
+        "privateIPAllocationMethod": {
+          "$ref": "./network.json#/definitions/IPAllocationMethod",
+          "description": "The private IP address allocation method."
+        },
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The reference to the subnet resource."
```
Comment (MyronFanQiu) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json:4819]: do we need specific name for the subnet?

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Please address validation errors

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/examples/NetworkVirtualApplianceListByResourceGroup.json [hunk 4]
```diff
@@ -32,11 +32,16 @@
               "virtualHub": {
                 "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualHubs/hub1"
               },
+    		      "virtualApplianceSites": [
```
Comment (majastrz) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/examples/NetworkVirtualApplianceListByResourceGroup.json:35]: >		 [](start = 4, length = 2)

nit - tabs and spaces mixed together. You have this problem in several examples.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json [hunk 5]
```diff
@@ -473,6 +814,148 @@
...
+        }
+      },
+      "description": "Network Virtual Appliance Sku Properties."
+    },
+    "BreakOutCategoryPolicies": {
+      "properties": {
+        "allow": {
+          "type": "boolean",
+          "readOnly": false,
+          "description": "Policy to allow."
```
[the remark this answers is not in the document]
Comment (majastrz) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json:881]: I am confused by this as well.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/examples/NetworkVirtualApplianceListByResourceGroup.json (hunk 4 above)
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/examples/NetworkVirtualApplianceListByResourceGroup.json]: Done. Made changes to convert tabs to spaces in all the files

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json (hunk 3 above)
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json:4819]: No. This is same as IpConfiguration in networkInterface.json. Subnet is generic name and we would need it to be as it is

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json (hunk 2 above)
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json]: Done. Thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json (hunk 1 above)
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/virtualWan.json]: Done. Thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json [hunk 6]
```diff
@@ -473,6 +814,148 @@
...
+        },
+        "optimize": {
+          "type": "boolean",
+          "readOnly": false,
+          "description": "Policy to Optimize."
+        },
+        "default": {
+          "type": "boolean",
+          "readOnly": false,
+          "description": "Default Policy."
```
[the remark this answers is not in the document]
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json]: Done. Thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json [hunk 7]
```diff
@@ -473,6 +814,148 @@
...
+      "properties": {
+        "allow": {
+          "type": "boolean",
+          "readOnly": false,
+          "description": "Policy to allow."
+        },
+        "optimize": {
+          "type": "boolean",
+          "readOnly": false,
+          "description": "Policy to Optimize."
```
[the remark this answers is not in the document]
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json]: Done. Thanks!

File: specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (naveenchekuri) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/networkVirtualAppliance.json]: Done. Thanks!

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by MyronFanQiu** [comment]
@naveenchekuri Please fix LintDiff issue. @number213 Please review it again. Thanks!

**Review by anton-evseev — APPROVED** [review]

**Pull request merged**